### ISSUE TITLE
AKS BYO CNI cluster with apiserver-vnet-integration. CLI ignores the IP provided in the --helm-set k8sServiceHost parameter

### ISSUE THREAD

**Issue by HummingMind** [cilium/cilium-cli#1788]
Hello,
Not sure if this is a bug or would be considered an existing feature improvement (I am leaning towards a bug):

**General Information**

- Version: cilium-cli: v0.15.0 compiled with go1.20.4 on linux/amd64
- k8s version: Client Version: v1.27.3,  Kustomize Version: v5.0.1, Server Version: v1.27.1
- Azure AKS BYO CNI, Public cluster with apiserver-vnet-integration enabled

**How to reproduce the issue**

1) I've created an AKS cluster with BYO CNI (so network plugin none), kube-porxy not installed (so I can use kubeProxyReplacement strict), and apiserver-vnet-integration enabled. This way I get a private endpoint IP (of a load balancer) to use for internal communication with the control plane (the api server). See [this link](https://techcommunity.microsoft.com/t5/core-infrastructure-and-security/public-and-private-aks-clusters-demystified/ba-p/3716838) for details. Scenario 3.

2) Installed the Gateway API CRDs.

3) Grabbed the (internal) endpoint IP for the next step by running:
`kubectl get endpoints -n default`

4) Ran the Cilium CLI to install Cilium (in Helm mode):
`cilium install --helm-set kubeProxyReplacement=strict --helm-set gatewayAPI.enabled=true --helm-set k8sServiceHost=internalEndpointIpFromStep3 --helm-set k8sServicePort=443 --version 1.14.0-rc.0 --helm-set hubble.relay.enabled=true --helm-set hubble.ui.enabled=true --helm-set envoy.enabled=true --helm-set azure.resourceGroup=myResourceGroupName`

Everything came up correctly, however, looking at the current value of k8sServiceHost it looks like the Cilium CLI ignored the parameter I specified and use the public FQDN DNS entry assigned by AKS to be able to reach the control plane externally. This should not be happening, as the whole idea of this type of setup is to have the nodes and the pods communicate with the control plane via the internal IP instead of over the Internet.

So, I had to run this to "fix" the issue:
`helm upgrade cilium cilium/cilium --namespace kube-system --version 1.14.0-rc.0 --reuse-values --set k8sServiceHost=internalEndpointIpFromStep3`

The short version of the bug is that the CLI should not be ignoring the value provided as the parameter.

Thank you!

PS: Should also mention this happens for kind and minikube clusters.
